File: software/examples/vbnet/ExampleInput.vb
Imports System
Imports Tinkerforge

Module ExampleInput
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your IO-16 Bricklet

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim io As New BrickletIO16(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current value from port A as bitmask
        Dim valueMaskA As Byte = io.GetPort("a"C)
        Console.WriteLine("Value Mask (Port A): " + Convert.ToString(valueMaskA, 2))

        ' Get current value from port B as bitmask
        Dim valueMaskB As Byte = io.GetPort("b"C)
        Console.WriteLine("Value Mask (Port B): " + Convert.ToString(valueMaskB, 2))

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleInput.pas
program ExampleInput;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletIO16;

type
  TExample = class
  private
    ipcon: TIPConnection;
    io: TBrickletIO16;
  public
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change XYZ to the UID of your IO-16 Bricklet }

var
  e: TExample;

procedure TExample.Execute;
var valueMaskA, valueMaskB: byte;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  io := TBrickletIO16.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get current value from port A as bitmask }
  valueMaskA := io.GetPort('a');
  WriteLn(Format('Value Mask (Port A): %d', [valueMaskA]));

  { Get current value from port B as bitmask }
  valueMaskB := io.GetPort('b');
  WriteLn(Format('Value Mask (Port B): %d', [valueMaskB]));

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
